.github/workflows/on.release.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: Release
+
+"on":
+  push:
+    tags:
+      - "v*.*.*"
+
+  workflow_dispatch:
+    inputs:
+      prerelease:
+        description: Pre-release
+        type: boolean
+        default: false
+      artifact:
+        description: Artifact
+        type: string
+        default: "capnp-cpp"
+      dry:
+        description: Dry-run
+        type: boolean
+        default: false
+
+permissions:
+  contents: read
+
+jobs:
+  release:
+    name: "Release: BCR"
+    uses: bazel-contrib/.github/.github/workflows/release_ruleset.yaml@v2
+    with:
+      release_files: capnp-cpp-*.tar.gz
```

Comment:
When a tag is pushed, `release-prep.sh` is run by this step, which then grabs the outputs described and publishes them on a Github Release as artifacts. This whole flow is provided by the `bazel-contrib` repository as a reusable workflow, paired with in-repo scripts/configs (i.e. `.bcr`, `.github/workflows/ci.bazelrc`, and so on).